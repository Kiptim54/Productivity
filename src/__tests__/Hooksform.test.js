import React from 'react';
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect'
import Counter from '../components/Hooksform';

afterEach(cleanup)
describe('learning and testing the counter component', () => {
    it('component should render', () => {
        const { container, getByText, getByTestId } = render(<Counter />);
        expect(container).toMatchSnapshot();
    });

    it("should have the counter number the on load", ()=>{
        const {getByTestId} = render(<Counter/>)
        expect(getByTestId('countertag')).toHaveTextContent("0")
    })

    it("should increment when button clicked", ()=>{
        const {getByTestId, getByText} = render(<Counter/>)
        const incrementButton = getByText("Increase")

        fireEvent.click(incrementButton)
        expect(getByTestId('countertag')).toHaveTextContent("1")
    })

    it("should decrement when button clicked", async ()=>{
        const {getByTestId, getByText} = render(<Counter/>)
        const decrementButton = getByText("Decrease")
        fireEvent.click(decrementButton)
        const reponse = await waitForElement(()=> getByText("-1"))
        expect(reponse).toHaveTextContent("-1")
    })
});
