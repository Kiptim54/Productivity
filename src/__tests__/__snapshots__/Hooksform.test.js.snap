// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`learning and testing the counter component component should render 1`] = `
<div>
  <div
    class="container"
  >
    <button>
      Increase
    </button>
    <p
      data-testid="countertag"
    >
      0
    </p>
    <button>
      Decrease
    </button>
  </div>
</div>
`;
